pennylane/tape/qnode.py

```diff
@@ -445,6 +444,30 @@ def __call__(self, *args, **kwargs):
             return __import__(res_type_namespace).numpy.squeeze(res)
         return __import__(res_type_namespace).squeeze(res)
 
+    def metric_tensor(self, *args, diag_approx=False, only_construct=False, **kwargs):
+        """Evaluate the value of the metric tensor.
+
+        Args:
+            args (tuple[Any]): positional arguments
+            kwargs (dict[str, Any]): auxiliary arguments
+            diag_approx (bool): iff True, use the diagonal approximation
+            only_construct (bool): Iff True, construct the circuits used for computing
+                the metric tensor but do not execute them, and return the tapes.
+
+        Returns:
+            array[float]: metric tensor
+        """
+        self.construct(args, kwargs)
+        metric_tensor_tapes, processing_fn = qml.tape.transforms.metric_tensor(
+            self.qtape, diag_approx=diag_approx
+        )

```

Comment:
Nice :) Like a diff rule

Reply:
That was my inspiration 😆 

Funnily enough, in https://arxiv.org/abs/2008.06517, a 4-term parameter-shift-like rule is provided for computing every metric tensor element. So you get the full metric tensor on hardware, but with more evaluations (`4p` evaluations for `p` parameters) than the block diagaonal approximation (`L` evaluations, where `L` is the number of layers)

We should add this at some point (too much work to do it now).